resources/aws_ec2.go

```diff
@@ -1038,6 +1075,91 @@ func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+// snsVerifySignature verifies that the post messages are genuine and originate
+// from amazon by checking if the signature is valid for the provided key and
+// message contents.
+func (obj *AwsEc2Res) snsVerifySignature(post postData) error {
+	// download and parse the signing certificate
+	cert, err := obj.snsGetCert(post.SigningCertURL)
+	if err != nil {
+		return errwrap.Wrapf(err, "error getting certificate")
+	}
+	// convert the message to canonical form
+	message := obj.snsCanonicalFormat(post)
+	// decode the message signature from base64
+	signature, err := base64.StdEncoding.DecodeString(post.Signature)
+	if err != nil {
+		return errwrap.Wrapf(err, "error decoding string")
+	}
+	// check the signature against the message
+	err = cert.CheckSignature(x509.SHA1WithRSA, message, signature)
```

Comment:
err := and group with the if please.

alternatively we could

return errwrap.Wrapf(cert.CheckSignature(x509.SHA1WithRSA, message, signature),  "error checking signature")

but that's too long and messy. Better to do the first fix and keep it obvious for crypto.